db/upgrade.php

```diff
@@ -64,6 +64,39 @@ function create_xapi_log_table($dbman, $tablename) {
     }
 }
 
+function create_logstoreid_and_type_columns($dbman) {
+
+    // Select table.
+    $table = new xmldb_table('logstore_xapi_log');
+
+    // Conditionally add field logstore_standard_log_id to logstore_xapi_log.
+    $field = new xmldb_field('logstore_standard_log_id', XMLDB_TYPE_INTEGER, '10', null, null, null, null);
+    if (!$dbman->field_exists($table, $field)) {
+        $dbman->add_field($table, $field);
+    }
+
+    // Conditionally add field type to logstore_xapi_log.
+    $field = new xmldb_field('type', XMLDB_TYPE_INTEGER, '10', null, null, null, null);
+    if (!$dbman->field_exists($table, $field)) {
+        $dbman->add_field($table, $field);
+    }
+
+    // Select table.
+    $table = new xmldb_table('logstore_xapi_failed_log');
+
+    // Conditionally add field logstore_standard_log_id to logstore_xapi_log.
+    $field = new xmldb_field('logstore_standard_log_id', XMLDB_TYPE_INTEGER, '10', null, null, null, null);
+    if (!$dbman->field_exists($table, $field)) {
+        $dbman->add_field($table, $field);
+    }
+
+    // Conditionally add field type to logstore_xapi_log.
+    $field = new xmldb_field('type', XMLDB_TYPE_INTEGER, '10', null, null, null, null);
+    if (!$dbman->field_exists($table, $field)) {
+        $dbman->add_field($table, $field);
+    }
+}
```

Comment:
there's no indexes defined on the new columns

Reply:
I have added simple indexes to install.xml and upgrade.php. Type will probably be queried but logstorestandardlogid will probably not be queried.